cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -692,7 +692,12 @@ pub mod pallet {
 				vfp.relay_parent_number,
 			));
 
-			Ok(PostDispatchInfo { actual_weight: Some(total_weight), pays_fee: Pays::No })
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
+				total_weight,
+				DispatchClass::Mandatory,
+			);
+
+			Ok(Pays::No.into())
```

Comment:
Nit: You can change the return type of the dispatchable to just `DispatchResult`.

Reply:
done in https://github.com/paritytech/polkadot-sdk/pull/7327/commits/e988d2b30a24c3b0f3d69b914d6dae8b8c54826a